Use public libpng accessors and typed SDL mapper arguments for libtcod 1.5.0.
No renderer or gameplay behavior is replaced.

--- a/src/sys_sdl_c.c
+++ b/src/sys_sdl_c.c
@@ -170,7 +170,7 @@
 	TCOD_sdl_renderer=renderer;
 }
 
-void TCOD_sys_map_ascii_to_font(asciiCode, fontCharX, fontCharY) {
+void TCOD_sys_map_ascii_to_font(int asciiCode, int fontCharX, int fontCharY) {
 	if ( asciiCode > 0 && asciiCode < TCOD_max_font_chars )
 		ascii_to_tcod[asciiCode] = fontCharX + fontCharY * fontNbCharHoriz;
 }
--- a/src/sys_sdl_img_png.c
+++ b/src/sys_sdl_img_png.c
@@ -65,7 +65,7 @@
 	if (info_ptr == NULL)
 	{
 		fclose(fp);
-		png_destroy_read_struct(&png_ptr, png_infopp_NULL, png_infopp_NULL);
+		png_destroy_read_struct(&png_ptr, NULL, NULL);
 		return NULL;
 	}
 
@@ -77,7 +77,7 @@
 	if (setjmp(png_jmpbuf(png_ptr)))
 	{
 		/* Free all of the memory associated with the png_ptr and info_ptr */
-		png_destroy_read_struct(&png_ptr, &info_ptr, png_infopp_NULL);
+		png_destroy_read_struct(&png_ptr, &info_ptr, NULL);
 		fclose(fp);
 		/* If we get here, we had a problem reading the file */
 		return NULL;
@@ -93,7 +93,7 @@
 	* adjustment), then you can read the entire image (including
 	* pixels) into the info structure with this call:
 	*/
-	//png_read_png(png_ptr, info_ptr, PNG_TRANSFORM_IDENTITY, png_voidp_NULL);
+	//png_read_png(png_ptr, info_ptr, PNG_TRANSFORM_IDENTITY, NULL);
 
 	// get info about the image
 	png_read_info(png_ptr,info_ptr);
@@ -112,7 +112,7 @@
 		&png_interlace_type,NULL,NULL);
 
 	// create the SDL surface
-	bitmap=TCOD_sys_get_surface(png_width,png_height,info_ptr->channels == 4);
+	bitmap=TCOD_sys_get_surface(png_width,png_height,png_get_channels(png_ptr, info_ptr) == 4);
 
 	// get row data
 	row_pointers=(png_bytep *)malloc(sizeof(png_bytep)*png_height);
@@ -124,7 +124,7 @@
 	png_read_image(png_ptr,row_pointers);
 
 	/* clean up after the read, and free any memory allocated - REQUIRED */
-	png_destroy_read_struct(&png_ptr, &info_ptr, png_infopp_NULL);
+	png_destroy_read_struct(&png_ptr, &info_ptr, NULL);
 	free(row_pointers);
 
 	/* close the file */
@@ -153,14 +153,14 @@
 	if (info_ptr == NULL)
 	{
 		fclose(fp);
-		png_destroy_write_struct(&png_ptr, png_infopp_NULL);
+		png_destroy_write_struct(&png_ptr, NULL);
 		return;
 	}
 
 	if (setjmp(png_jmpbuf(png_ptr)))
 	{
 		/* Free all of the memory associated with the png_ptr and info_ptr */
-		png_destroy_read_struct(&png_ptr, &info_ptr, png_infopp_NULL);
+		png_destroy_read_struct(&png_ptr, &info_ptr, NULL);
 		fclose(fp);
 		/* If we get here, we had a problem reading the file */
 		return;
